src/process.c

```diff
@@ -179,6 +183,14 @@ void process_wakeup( struct list *q )
 	}
 }
 
+void process_reap_all()
+{
+	struct process *p;
+	while((p = (struct process*)list_pop_head(&grave_list))) {
+        pagetable_delete(p->pagetable);
```

Comment:
Yes, that's the right idea, but there is (and will be) more to cleaning up a process than deleting the page table.
Have this fn call `process_delete` which should delete all of its sub-elements and then `kfree(p)`